src/adapted_grid.jl

```diff
@@ -119,10 +120,7 @@ function adapted_grid(f, minmax::Tuple{Real, Real}; max_recursions = 7)
                 end
             else
                 new_xs[i + k] = xs[i]
-                # Don't evaluate function at end points
-                if !(i == 1 || i == n_points)
-                    new_fs[i + k] = fs[i]
-                end
+                new_fs[i + k] = fs[i]
```

Comment:
The end points have `NaN` values which should propagate. Otherwise they remain zero, which messes up the scaling.